1) +

Gaydarenko opened the pull request:
1) +
2)

Gaydarenko committed:
1) +
2)

aeksei commented on Offset.py:
```diff
@@ -6,3 +6,4 @@
     a = a // 1.7        # O(1)
 out.merge_sort()        # O(nlog(n))
 
+2)
```
А итоговая сложность какая получается?
и вставки в список немного другая сложность, Python list, не как классический список работает, а смесь массива и списка
- https://www.yuripetrov.ru/edu/python/ch_06_01.html#id13
- https://wiki.python.org/moin/TimeComplexity

aeksei reviewed (commented):
@Gaydar-Kanub

Gaydarenko committed:
1) +
2) +
3)

aeksei reviewed (commented):
1 и 2 +

Gaydarenko committed:
nothing interesting

Gaydarenko committed:
3) +

aeksei reviewed (commented):
3 +

Gaydarenko committed:
7) +

aeksei commented on Offset.py:
```diff
@@ -1,8 +1,78 @@
+# Целесообразно использовать сортировку слиянием, т.к. много однообразных значений.
+# А сортировка слиянием обладает свойством устойчивости.
+#
+# Но для данного случая я бы выбрал сортировку подсчётом (counting sort)
```
сортировка подсчетом подходит

aeksei reviewed (commented):
7 +

Gaydarenko committed:
6) +

Сделал небольшие правки по предыдущему коду для удовлетворения требований PEP8 - чтобы PyCharm не ругался

aeksei commented on Offset.py:
```diff
@@ -66,6 +70,36 @@ def connect_comp(g: nx.Graph):
+"""
+
+
+def sum_time(queue: list) -> bool:
+    """
+    Функция проверяет возможность удовлетворения ВСЕХ поданных заявок - сравнивает время окончания каждой
+    с временем начала следующей.
+    :param queue: список заявок
+    :return: bool
+    """
+    for i in range(len(queue) - 1):
+        if queue[i][1] > queue[i + 1][0]:
+            return False
+    return True
```
а здесь же не гарантируется, что заявки отсортированы
если попадется вот такой случай (10, 12), (8, 11)?

Gaydarenko replied:
Подправил решение (переделал полностью)

Gaydarenko committed:
7) реализовал решение

Gaydarenko committed:
5) +

Gaydarenko committed:
6) Добавлена сортировка

aeksei reviewed (commented):
Задача с консунсус рядами тоже зачтена

Gaydarenko committed:
4) +

Gaydarenko committed:
fix task № 2
Переделал задачу, т.к. в не
